rfcs/text/0000-matrix-service.md

```diff
@@ -0,0 +1,406 @@
+
+## Authors
+
+[Calvin Nhieu](cnhieu@chanzuckerberg.com)
+[Marcus Kinsella](mkinsella@chanzuckerberg.com)
+
+## Shepherd
+ [Brian Raymor](braymor@chanzuckerberg.com)
+
+## Motivation
+
+The Human Cell Atlas (HCA) Data Coordination Platform (DCP) ingests, processes, and stores single-cell data, making that data easily available to a wide audience of researchers. The single-cell data type of greatest interest to most researchers is the expression matrix, the cell-by-gene expression values that are the starting point for many downstream analyses. To support this most commonly requested data type, the DCP offers a specialized interface, the DCP matrix service, which enables investigators to easily access and perform integrative analysis across HCA expression data.
+
+### User Stories
+
+As a data consumer, I would like to access gene expression data from the DCP so that I can perform downstream integrative analysis across HCA experiments.
+
+- I would like to be able to receive an expression matrix comprised of cells that satisfy a set of metadata conditions.
+- I would like to receive metadata providing the experimental and scientific context about the cells in my matrix.
+- I would like to receive my matrix in a format that is compatible with standard single-cell analysis tools.
+
+## Scientific "guardrails"
+
+*Describe recommended or mandated review from HCA Science governance to ensure that the RFC addresses the needs of the scientific community.*
+
+## Detailed Design
+
+### Overview
+
+The Matrix Service enables on-the-fly expression matrix generation via a [RESTful API](https://matrix.data.humancellatlas.org/). Through user studies and iterative development over the past year, four major requirements of matrices have been identified and must be met. One, matrices must enable integrative analysis across data in the HCA and the Matrix Service determines that this integration should be at the granularity of individual cells. Second, matrices should include rich metadata that users are interested in so that they can better understand the data they're working with. Third, matrices should be compatible with downstream single cell analysis tools, for example in `Python` and `R`, to further enable downstream analysis of HCA data by the wider community. Finally, performance of serving matrices should be prioritized as the scale of HCA data is anticipated to reach or exceed the order of 10<sup>7</sup> cells. The Matrix Service API is designed to fulfill these requirements in order to provide access to HCA data in a format that enables and encourages downstream integrative analysis.
```

Comment:
- Could you break out the 4 requirements into bullet points please?
- Matrix Service didn't determine that the integration should be at the cell-granularity.. it was the users, right? Could reword to "matrices must enable cell-level granularity integrative analysis across data in the HCA"
- On your second point, might swap the word "should" to "must" since it was a requirement.
- On your third point, "for example in Python and R" isn't very descriptive of the analysis tools. I think these could be "Seurat and Scanpy"? (@mckinsel to confirm)
- On your fourth point, instead of using 10^7, maybe just write out "tens of millions" for clarity.